util.go

```diff
@@ -170,3 +171,29 @@ func Equal(a, b interface{}) bool {
 	}
 	return false
 }
+
+// Walk is a filepath.Walk helper which follows symlinks and always returns the
+// path of the file or directory that is referenced in the link.
+func Walk(root string, walkFn filepath.WalkFunc) error {
+	var symWalkFn = func(path string, info os.FileInfo, err error) error {
+		var isLink bool
+		// get the target if it is a symlink
+		if err == nil && info.Mode()&os.ModeSymlink == os.ModeSymlink {
+			isLink = true
+			path, err = filepath.EvalSymlinks(path)
+			if err != nil { // abort if we can't evaluate the symlink (e.g. a dangling symlink)
+				return err
+			}
+			info, err = os.Lstat(path)
+		}
+		if err := walkFn(path, info, err); err != nil {
+			return err // abort on walkFn error too (same behavior as filepath.Walk)
+		}
+		// recurse into symlinked directories
+		if err == nil && info.IsDir() && isLink {
+			return Walk(path, walkFn)
```

Comment:
this code cannot detect loops in the directory structure -> so if loops do exist it runs the risk of going into infinite recursion.

for that I wrote some loop detection code & another alternative implementation for the whole code. [it mostly copies from filepath.Walk]

https://gist.github.com/pavanka/2cd9f21639446d53b7b6